src/python/pants/engine/target.py

```diff
@@ -878,13 +882,20 @@ class TransitiveTargetsRequest:
     """
 
     roots: Tuple[Address, ...]
-    include_special_cased_deps: bool
+    should_traverse_deps_predicate: ShouldTraverseDepsPredicate
 
     def __init__(
-        self, roots: Iterable[Address], *, include_special_cased_deps: bool = False
+        self,
+        roots: Iterable[Address],
+        *,
+        should_traverse_deps_predicate: ShouldTraverseDepsPredicate | None = None,
```

Comment:
Instead of defaulting inside the body, couldn't we just default here in the signature?

Reply:
That caused a bunch of type errors within PyCharm. I can try and see if mypy accepts it.

Reply:
It should be 100% legal/valid

Reply:
Addressed in #19272